kernel/src/schema/mod.rs

```diff
@@ -567,14 +852,7 @@ impl<'de> Deserialize<'de> for StructType {
         Self: Sized,
     {
         let helper = StructTypeSerDeHelper::deserialize(deserializer)?;
-        Ok(Self {
-            type_name: helper.type_name,
-            fields: helper
-                .fields
-                .into_iter()
-                .map(|f| (f.name.clone(), f))
-                .collect(),
-        })
+        Ok(StructType::new_unchecked(helper.fields))
```

Comment:
this seems unsafe? shouldn't we propagate the error?

Reply:
I initially considered this as a micro-optimization, assuming that any `StructType` that was once initialized and serialized _should_ be valid. But re-checking with `try_new` might be the safer option here if you are not concerned about re-running the checks in `try_new`.

Reply:
I talked to @scovich, who agreed that my initial idea is potentially dangerous. I changed it.